packages/itwinui-react/src/core/SearchBox/SearchBox.tsx

```diff
@@ -0,0 +1,398 @@
+    onToggle?.(false);
+    openButtonRef.current && openButtonRef.current.focus();
+  };
+
+  const onOpen = () => {
+    setLocalExpanded(true);
+    onToggle?.(true);
+    inputRef.current && inputRef.current.focus();
+  };
+
+  return (
+    <SearchBoxContext.Provider
+      value={{
+        size,
+        disabled,
+        onOpen,
+        onClose,
+        inputRef,
+        inputId,
+        setInputId,
+        openButtonRef,
+      }}
+    >
+      <InputFlexContainer
+        ref={ref}
+        className={cx({ 'iui-expandable-searchbox': expandable }, className)}
+        data-iui-size={size}
+        disabled={disabled}
+        role='searchbox'
+        data-iui-expanded={localExpanded}
+        {...rest}
+      >
+        {!expandable &&
+          (children ?? (
+            <>
+              <SearchBoxInput {...inputProps} />
+              <SearchBoxIcon />
+            </>
+          ))}
+        {expandable && (
+          <>
+            <div className='iui-searchbox-collapsed'>
+              {collapsedState ?? <SearchBoxOpenButton {...openButtonProps} />}
+            </div>
+            <div className='iui-searchbox-expanded'>
+              {children ?? (
+                <>
+                  <SearchBoxInput {...inputProps} />
+                  <SearchBoxCloseButton {...closeButtonProps} />
+                </>
+              )}
+            </div>
+          </>
+        )}
+      </InputFlexContainer>
+    </SearchBoxContext.Provider>
+  );
+}) as PolymorphicForwardRefComponent<
+  'div',
+  SearchBoxOwnProps & InputFlexContainerProps
+>;
+
+export type SearchBoxProps = PolymorphicComponentProps<
+  'div',
+  SearchBoxOwnProps & InputFlexContainerProps
+>;
+
+// ----------------------------------------------------------------------------
+export type SearchBoxIconProps = PolymorphicComponentProps<'span', IconProps>;
+
+const SearchBoxIcon = React.forwardRef((props, ref) => {
+  const { size, className, children, ...rest } = props;
+  const { size: sizeContext } = useSafeContext(SearchBoxContext);
+
+  return (
+    <Icon
+      aria-hidden
+      className={cx('iui-svg-icon', 'iui-search-icon', className)}
+      size={size ?? sizeContext}
+      ref={ref}
+      {...rest}
+    >
+      {children ?? <SvgSearch />}
+    </Icon>
+  );
+}) as PolymorphicForwardRefComponent<'span', IconProps>;
+
+// ----------------------------------------------------------------------------
+
+export type SearchBoxInputProps = {
+  label?: string;
+} & React.ComponentProps<'input'>;
+
+const SearchBoxInput = React.forwardRef(
+  (props: SearchBoxInputProps, ref: React.Ref<HTMLInputElement>) => {
+    const {
+      className,
+      label = 'Search input',
+      id: idProp,
+      disabled: disabledProp,
+      ...rest
+    } = props;
+
+    const { inputId, setInputId, disabled, inputRef } =
+      useSafeContext(SearchBoxContext);
+
+    if (idProp && idProp !== inputId) {
+      setInputId(idProp);
+    }
```

Comment:
this should be moved into useEffect. see #1236

Reply:
Added to useEffect